pkg/agent/proxy/proxier.go

```diff
@@ -422,18 +459,27 @@ func NewProxier(
 	hostname string,
 	informerFactory informers.SharedInformerFactory,
 	ofClient openflow.Client,
+	supportedEndpointSliceVersion map[string]bool,
 	isIPv6 bool) *proxier {
 	recorder := record.NewBroadcaster().NewRecorder(
 		runtime.NewScheme(),
 		corev1.EventSource{Component: componentName, Host: hostname},
 	)
 	metrics.Register()
 	klog.Infof("Creating proxier with IPv6 enabled=%t", isIPv6)
+
+	enableEndpointSlice := false
+	// only EndpointSlice V1Beta1 is supported for now.
+	if supportedEndpointSliceVersion["v1beta1"] && features.DefaultFeatureGate.Enabled(features.EndpointSlice) {
+		enableEndpointSlice = true
+	}
+	klog.Infof("Creating proxier with EndpointSlice enabled=%t", enableEndpointSlice)
```

Comment:
The result of enable EndpointSlice is decided by two parts. One is feature gate, the other is the supported EndpointSlice API versions. So far, only v1beta1 is supported. @antoninbas